### PR TITLE
ZEPPELIN-1048: Pandas support for python interpreter

### PR DESCRIPTION
**Pull request by bzz**
### What is this PR for?
Display Pandas DataFrame using Zeppelin's Table Display system.

### What type of PR is it?
Feature

### Todos
* [x] fix NPE in logs on empty paragraph execution
* [x] matplotlib: refactor `zeppelin_show(plt)` -> `z.show(plt)`
* [x] pandas: support `z.show(df)`
* [x] update docs

### What is the Jira issue?
[ZEPPELIN-1048](https://issues.apache.org/jira/browse/ZEPPELIN-1048)

### How should this be tested?
"Zeppelin Tutorial: Python - matplotlib basic" should work, and

```python
import pandas as pd
rates = pd.read_csv("bank.csv", sep=";")
z.show(rates)
```
### Screenshots (if appropriate)
screen shot 2016-06-23 at 10 29 00

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? Yes

### CODE REVIEW

**Comment by bzz** [comment]
It is ready to be merged now.

TODOs will be addressed in different PR, after #1063 is merged

**Comment by Leemoonsoo** [comment]
I saw https://issues.apache.org/jira/browse/ZEPPELIN-1048 marked 0.7.0 as a fix version.

Not a strong opinion, and i don't think it should be blocker for 0.6 release. But If we're going to change api from `zeppelin_show(plt)` to `z.show(plt)`, how about apply this api change to 0.6 branch?

While `zeppelin_show()` introduced recently, now API change cost is not high. But once 0.6.0 released and people starts use `zeppelin_show()`, API change cost will be increased.

File: docs/interpreter/python.md [hunk 1]
```diff
@@ -83,13 +82,25 @@ zeppelin_show function can take optional parameters to adapt graph width and hei
 
  ```python
 %python
-zeppelin_show(plt,width='50px')
-zeppelin_show(plt,height='150px')
+zeppelin_show(plt, width='50px')
+zeppelin_show(plt, height='150px')
```
Comment (minahlee) [inline docs/interpreter/python.md]: This also need to be changed to `z.show()` I guess?

File: python/src/main/resources/bootstrap.py [hunk 2]
```diff
@@ -56,48 +57,102 @@ def help():
...
-            style += 'height:'+height
-    print("%html <div style='" + style + "'>" + img.read() + "<div>")
+    print ('''<pre>z.show(plt,width='50px')
+z.show(plt,height='150px') </pre></div>''')
+    print ('<h3>Pandas DataFrame</h3>')
+    print """
+<div>The interpreter can visualize Pandas DataFrame
+with the function z.show()
+<pre>
+import matplotlib.pyplot as plt
```
Comment (minahlee) [inline python/src/main/resources/bootstrap.py]: `import pandas as pd` instead `import matplotlib.pyplot as plt`?

**Comment by minahlee** [comment]
I left few notes on documentation and `help() ` function. Looks good to me.

File: docs/interpreter/python.md (hunk 1 above)
Comment (bzz) [inline docs/interpreter/python.md]: Good catch, of course it should. Commit addressing it was lost on rebasing.

File: python/src/main/resources/bootstrap.py (hunk 2 above)
Comment (bzz) [inline python/src/main/resources/bootstrap.py]: Thanks for catching!

**Comment by bzz** [comment]
Thank you guys for the prompt reviews!

@Leemoonsoo agree on API change and I'm glad to have it merged in 0.6, updated JIRA
@minahlee review addressed in 3b1ad36

If there is no more discussion, going to merge it to *master* and *branch-0.6*

**Comment by bzz** [comment]
cherry-picking to branch-0.6 fails with merge conflicts as master and branch-0.6 diverged

**Comment by bzz** [comment]
done in 97f25e2

**Pull request closed without merging**